data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -280,9 +220,17 @@ impl TraceExporter {
         trace_count: usize,
     ) -> Result<AgentResponse, TraceExporterError> {
         self.check_agent_info();
+
         match self.input_format {
             TraceExporterInputFormat::Proxy => self.send_proxy(data.as_ref(), trace_count),
-            TraceExporterInputFormat::V04 => self.send_deser_ser(data),
+            TraceExporterInputFormat::V04 => match msgpack_decoder::v04::from_slice(data) {
```

Comment:
non-blocking question - Should we eventually encapsulate some of this logic in the decoder? You pass a TraceCollection and the format and it figures out which particular version's decode to call?